packages/experimental/navigation/otel/hooks/useTrace.ts

```diff
@@ -0,0 +1,39 @@
+import {trace, Tracer, TracerProvider} from '@opentelemetry/api';
+import {MutableRefObject, useEffect, useRef} from 'react';
+
+const TRACER_DEFAULT = {
+  name: 'default',
+  version: '1.0',
+};
+
+interface ConfigArgs {
+  name?: string;
+  version?: string;
+}
+
+type TracerRef = MutableRefObject<Tracer | null>;
+
+const useTrace = (
+  {name, version}: ConfigArgs = {},
+  provider: TracerProvider,
+): TracerRef => {
+  // otel
+  const tracerRef = useRef<Tracer | null>(null);
+
+  // using the layout effect to make sure the tracer is initialized before the component is rendered
+  useEffect(() => {
```

Comment:
given the comment was this meant to be using `useLayoutEffect`?

Reply:
yes, I changed it for testing purposes but I just added `useLayoutEffect` back